test/e2e/proxy_test.go

```diff
@@ -0,0 +1,184 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"encoding/hex"
+	"fmt"
+	"net"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func proxyEnabled(data *TestData) (bool, error) {
+	key := "resubmit(,40),resubmit(,41)"
+	agentName, err := data.getAntreaPodOnNode(masterNodeName())
+	if err != nil {
+		return false, err
+	}
+	table31Output, _, err := data.runCommandFromPod(metav1.NamespaceSystem, agentName, "antrea-agent", []string{"ovs-ofctl", "dump-flows", defaultBridgeName, "table=31"})
+	return strings.Contains(table31Output, key), err
+}
+
+func TestProxyServiceSessionAffinity(t *testing.T) {
+	skipIfProviderIs(t, "kind", "#881 Does not work in Kind, needs to be investigated.")
+	data, err := setupTest(t)
+	if err != nil {
+		t.Fatalf("Error when setting up test: %v", err)
+	}
+	defer teardownTest(t, data)
+
+	if enabled, err := proxyEnabled(data); err != nil {
```

Comment:
maybe define `skipIfProxyDisabled(t, data)` in this file to reduce verbosity of this test and the subsequent ones

Reply:
Done.